### ISSUE TITLE
dotnet command not found: 1.0.1 install on clean macOS Sierra

### ISSUE THREAD

**Issue by leecow** [dotnet/cli#4225]
On macOS Sierra, (fresh install) the latest installer didn't set the correct path - dotnet in terminal returned command not found

OS version
macOS Sierra
Repro steps
Fresh install of macOS without prior dotnet core
Repro code (if you can share it / is relevant)
Error messages
command not found
Whether this worked before
Method of installing .NET Core
package installer

Ran

ln -s /usr/local/share/dotnet/dotnet /usr/local/bin

and started working

/cc @h3smith @blackdwarf

**Comment by blackdwarf** [comment]
@leecow can you check if you have `dotnet` in `/etc/paths.d/` location?

**Comment by leecow** [comment]
@blackdwarf I haven't seen this repro. @richlander can you have a look at the machine where this was repro'd?

**Comment by the-darkvoid** [comment]
Can confirm this issue on a fresh macOS Sierra installation on a MacBook Pro.
Running .NET Core installer does not make the `dotnet` command available in the terminal.

@blackdwarf,
Path `/etc/paths.d` does not exist at all.
`/etc/paths` exists as a file containing:
* /usr/local/bin
* /usr/bin
* /bin
* /usr/sbin
* /sbin

**Comment by KiltedViking** [comment]
F.Y.I.: I had the same problem on one of my three Macs. The difference between failing Mac (2011) and others, should it help:
- I'm on the beta program track, i.e. just got latest beta for macOS 10.12.1.
- I don't have XCode installed - only command line tool (for Homebrew, etc.).

They are all running Sierra and the other computers are a Mac Mini (2012; never reinstalled "clean") and a Macbook Pro (less than 1 year old). I installed .NET Core last year, i.e. "DNVM", on all of them.

I added link and I could run the command dotnet.

**Comment by blackdwarf** [comment]
@the-darkvoid thanks for that info. Can you try the following please and let me know if it fixes it:

```console
mkdir -p /etc/paths.d
echo /usr/local/share/dotnet > /etc/paths.d/dotnet
```

You may need to `sudo` both of the commands. This mimics the behavior of our postinstall step in the CLI installer. After these commands are done, open up a new terminal window and see if the command is on the $PATH now. (Of course, if you already symlinked it, please remove the symlink.)

Thanks much for your help!

**Comment by the-darkvoid** [comment]
@blackdwarf,

Creating the `paths.d` entry manually as suggested makes the `dotnet` command work.
Question remains is why the installer was unable to create these `paths.d` entries automatically.

Let me know if I need to dump some logs from the installer process.

**Comment by blackdwarf** [comment]
@the-darkvoid because the installer assumes that the `/etc/paths.d` exists. It did, I believe, on El Capitan. That means that we have an issue for the post-install step.

Any logs you can provide would be good. I guess this needs to go into the "known issues" document for Sierra.

**Comment by richjoslin** [comment]
The "sudo mkdir" works for me, but then the "sudo echo line" gives me this for some reason (tried with and without sudo):
`-bash: /etc/paths.d/dotnet: Permission denied`

**Comment by blackdwarf** [comment]
@richjoslin what is in the `/etc/paths.d/dotnet` file?

**Comment by richjoslin** [comment]
The dotnet file contains:
`/usr/local/share/dotnet`
It was owned by my user. I chown'd it to root, chmod to 0777, but that didn't help either.

**Comment by blackdwarf** [comment]
@richjoslin hmm, this is interesting. You are using bash? Are you setting the path anywhere manually in your `.bash_profile` or `.bashrc`?

**Comment by richjoslin** [comment]
I wasn't, but then I created a .bash_profile that adds that to my $PATH and all seems to be well now.

**Comment by blackdwarf** [comment]
This should be fixed with the dotnet/core-setup#362 change. If this problem persists, please reopen this issue.
